### PR TITLE
Create interface methods for GIDVerifyAccountDetail. 

### PR DESCRIPTION
**Pull request by brnnmrls**
Similar to GIDSignIn's interface, three methods are being added to GIDVerifyAccountDetail in order to start an interactive verify flow on iOS. Creating these methods also leads to creating the skeleton for GIDVerifiableAccountDetail and GIDVerifiedAccountDetailResult.

### CODE REVIEW

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 1]
```diff
@@ -15,6 +15,81 @@
...
+/// Starts an interactive verify flow on iOS.
+///
+/// The completion will be called at the end of this process.  Any saved verify state will be
+/// replaced by the result of this flow.
+///
+/// @param accountDetails A list of verifiable account details.
+/// @param presentingViewController The view controller used to present `SFSafariViewController` on
+///     iOS 9 and 10 and to supply `presentationContextProvider` for `ASWebAuthenticationSession` on
+///     iOS 13+.
+/// @param completion The optional block that is called on completion.  This block will
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:45]: Maybe "The optional block called asynchronously on the main queue upon completion."

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 2]
```diff
@@ -15,6 +15,81 @@
...
+///     be called asynchronously on the main queue.
+- (void)verifyAccountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails 
+         presentingViewController:(UIViewController *)presentingViewController 
+                       completion:(nullable void (^)(GIDVerifiedAccountDetailResult *_Nullable verifyResult, 
+                                                   NSError *_Nullable error))completion;
+
+/// Starts an interactive verify flow on iOS using the provided hint.
+///
+/// The completion will be called at the end of this process.  Any saved verify state will be
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:54]: s/verify/verification

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 3]
```diff
@@ -15,6 +15,81 @@
...
+NS_ASSUME_NONNULL_BEGIN
+
+/// This class is used to verify a user's Googleaccount details.
 @interface GIDVerifyAccountDetail : NSObject
+
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
+
+/// Starts an interactive verify flow on iOS.
+///
+/// The completion will be called at the end of this process.  Any saved verify state will be
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:38]: s/verify/verification

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 4]
```diff
@@ -15,6 +15,81 @@
...
+/// replaced by the result of this flow.
+///
+/// @param accountDetails A list of verifiable account details.
+/// @param presentingViewController The view controller used to present `SFSafariViewController` on
+///     iOS 9 and 10 and to supply `presentationContextProvider` for `ASWebAuthenticationSession` on
+///     iOS 13+.
+/// @param hint An optional hint for the authorization server, for example the user's ID or email
+///     address, to be prefilled if possible.
+/// @param completion The optional block that is called on completion.  This block will
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:63]: See above comment for a suggestion on making this simpler.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 5]
```diff
@@ -15,6 +15,81 @@
...
+///     iOS 9 and 10 and to supply `presentationContextProvider` for `ASWebAuthenticationSession` on
+///     iOS 13+.
+/// @param completion The optional block that is called on completion.  This block will
+///     be called asynchronously on the main queue.
+- (void)verifyAccountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails 
+         presentingViewController:(UIViewController *)presentingViewController 
+                       completion:(nullable void (^)(GIDVerifiedAccountDetailResult *_Nullable verifyResult, 
+                                                   NSError *_Nullable error))completion;
+
+/// Starts an interactive verify flow on iOS using the provided hint.
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:52]: Ditto remove "iOS."

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 6]
```diff
@@ -15,6 +15,81 @@
...
+///     address, to be prefilled if possible.
+/// @param completion The optional block that is called on completion.  This block will
+///     be called asynchronously on the main queue.
+- (void)verifyAccountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails 
+         presentingViewController:(UIViewController *)presentingViewController 
+                             hint:(nullable NSString *)hint
+                       completion:(nullable void (^)(GIDVerifiedAccountDetailResult *_Nullable verifyResult, 
+                                                   NSError *_Nullable error))completion;
+
+/// Starts an interactive verify flow on iOS using the provided hint and additional scopes.
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:71]: Ditto remove "iOS."

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 7]
```diff
@@ -15,6 +15,81 @@
...
+/// The completion will be called at the end of this process.  Any saved verify state will be
+/// replaced by the result of this flow. 
+///
+/// @param accountDetails A list of verifiable account details.
+/// @param presentingViewController The view controller used to present `SFSafariViewController` on
+///     iOS 9 and 10.
+/// @param hint An optional hint for the authorization server, for example the user's ID or email
+///     address, to be prefilled if possible.
+/// @param additionalScopes An optional array of scopes to request in addition to the basic profile scopes.
+/// @param completion The optional block that is called on completion.  This block will
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:82]: Same comment about simplifying this doc.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 8]
```diff
@@ -15,6 +15,81 @@
...
+///     iOS 13+.
+/// @param hint An optional hint for the authorization server, for example the user's ID or email
+///     address, to be prefilled if possible.
+/// @param completion The optional block that is called on completion.  This block will
+///     be called asynchronously on the main queue.
+- (void)verifyAccountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails 
+         presentingViewController:(UIViewController *)presentingViewController 
+                             hint:(nullable NSString *)hint
+                       completion:(nullable void (^)(GIDVerifiedAccountDetailResult *_Nullable verifyResult, 
+                                                   NSError *_Nullable error))completion;
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:69]: Nit: Align `NSError` with `GIDVerifiedAccountDetailResult`.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 9]
```diff
@@ -15,6 +15,81 @@
...
+///     iOS 9 and 10.
+/// @param hint An optional hint for the authorization server, for example the user's ID or email
+///     address, to be prefilled if possible.
+/// @param additionalScopes An optional array of scopes to request in addition to the basic profile scopes.
+/// @param completion The optional block that is called on completion.  This block will
+///     be called asynchronously on the main queue.
+- (void)verifyAccountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails 
+         presentingViewController:(UIViewController *)presentingViewController 
+                             hint:(nullable NSString *)hint
+                          additionalScopes:(nullable NSArray<NSString *> *)additionalScopes
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:87]: Nit: Align on colon.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 10]
```diff
@@ -15,6 +15,81 @@
...
+                       completion:(nullable void (^)(GIDVerifiedAccountDetailResult *_Nullable verifyResult, 
+                                                   NSError *_Nullable error))completion;
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:89]: Nit: Align `NSError` with `GIDVerifiedAccountDetailResult`.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 11]
```diff
@@ -15,6 +15,81 @@
...
+/// @param accountDetails A list of verifiable account details.
+/// @param presentingViewController The view controller used to present `SFSafariViewController` on
+///     iOS 9 and 10 and to supply `presentationContextProvider` for `ASWebAuthenticationSession` on
+///     iOS 13+.
+/// @param completion The optional block that is called on completion.  This block will
+///     be called asynchronously on the main queue.
+- (void)verifyAccountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails 
+         presentingViewController:(UIViewController *)presentingViewController 
+                       completion:(nullable void (^)(GIDVerifiedAccountDetailResult *_Nullable verifyResult, 
+                                                   NSError *_Nullable error))completion;
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:50]: Nit: Align `NSError` with `GIDVerifiedAccountDetailResult`.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 12]
```diff
@@ -15,6 +15,81 @@
...
+/// replaced by the result of this flow.
+///
+/// @param accountDetails A list of verifiable account details.
+/// @param presentingViewController The view controller used to present `SFSafariViewController` on
+///     iOS 9 and 10 and to supply `presentationContextProvider` for `ASWebAuthenticationSession` on
+///     iOS 13+.
+/// @param completion The optional block that is called on completion.  This block will
+///     be called asynchronously on the main queue.
+- (void)verifyAccountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails 
+         presentingViewController:(UIViewController *)presentingViewController
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:48]: Nit: Align colons.

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifyAccountDetail.m [hunk 13]
```diff
@@ -16,5 +16,34 @@
...
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiedAccountDetailResult.h"
+
 @implementation GIDVerifyAccountDetail
+
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
+
+- (void)verifyAccountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails
+    presentingViewController:(UIViewController *)presentingViewController
+                  completion:(nullable void (^)(GIDVerifiedAccountDetailResult *_Nullable verifyResult, NSError *_Nullable error))completion {
+    // TODO: Implement this method.
```
Comment (mdmathias) [inline GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifyAccountDetail.m:29]: In general, it's a good practice to link TODOs to an issue on the repo. I know we have internal tracking for this work, but we can talk about creating visibility here as well.

**Review by mdmathias — CHANGES_REQUESTED** [review]

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+/// Helper object used to hold the enumeration representing a list of
```
[the remark this answers is not in the document]
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:19]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiedAccountDetailResult.h [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+/// A helper object that contains the result of a successful verification or addScopes flow.
```
[the remark this answers is not in the document]
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiedAccountDetailResult.h:19]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 16]
```diff
@@ -15,6 +15,81 @@
...
+#elif __has_include(<AppKit/AppKit.h>)
+#import <AppKit/AppKit.h>
+#endif
+
+@class GIDVerifiableAccountDetail;
+@class GIDVerifiedAccountDetailResult;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// This class is used to verify a user's Googleaccount details.
```
[the remark this answers is not in the document]
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:31]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h [hunk 17]
```diff
@@ -15,6 +15,81 @@
...
 @interface GIDVerifyAccountDetail : NSObject
+
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
+
+/// Starts an interactive verify flow on iOS.
```
[the remark this answers is not in the document]
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:36]: Make sense, done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 1 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:45]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 2 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:54]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 3 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:38]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 5 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:52]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 6 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:71]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 4 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:63]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 7 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:82]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 8 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:69]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 9 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:87]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 10 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:89]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 11 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:50]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h (hunk 12 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h:48]: Done.

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifyAccountDetail.m (hunk 13 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifyAccountDetail.m]: Sounds good! I created the [issue 383](https://github.com/google/GoogleSignIn-iOS/issues/383) and added the issue number to the TODO. Let me know if there's a better way of linking the issue or if I should be adding in the entire link.

**Review by mdmathias — APPROVED** [review]

**Pull request merged**